### ISSUE TITLE
Code does not compile with Qt4

### ISSUE THREAD

**Issue by chraibi** [JuPedSim/jpseditor#46]
In Gitlab by @chraibi on Nov 4, 2015, 13:27 [[origin](https://gitlab.version.fz-juelich.de/jupedsim/jpseditor/issues/34)]

first problem is 

```
ui->classBox->setCurrentText(type);    // works with Qt5 only
```

Other problems are related to `QStringRef`

According to the [documentation](http://doc.qt.io/qt-4.8/qstringref.html)
this class has no member names `toInt` or `toFloat`

What should we do now? Stick with qt5 and ignore qt4? 
(hard decision)

**Comment by chraibi** [comment]
In Gitlab by @chraibi on Dec 17, 2016, 08:38

closed

**Comment by chraibi** [comment]
In Gitlab by @chraibi on Dec 17, 2016, 08:38

_Original comment by chraibi on 04 Nov 2015, 12:55_

---

mentioned in commit 072c17f7227bde3800daff7b4625ebb8cf4309b1

**Comment by chraibi** [comment]
In Gitlab by @chraibi on Dec 17, 2016, 08:38

_Original comment by chraibi on 04 Nov 2015, 12:55_

---

second problem solved.

**Comment by chraibi** [comment]
In Gitlab by @chraibi on Dec 17, 2016, 08:38

_Original comment by andresen on 08 Jan 2016, 11:03_

---

I changed the function ShowRoomType and omitted the above mentioned line. 
However, as I wanted to compile with qt4 I had to use the old MinGW 4.4 as newer compilers are not compatible with the newest qt 4 version, namely 4.8.4.. MinGW 4.4 does not entirely support c++11 functionality. Thus, I'm not sure if we are still able or rather want to keep qt 4 support if that means that certain c++11 function may not be used.
